proposal/TommyLin/README.rst

```diff
@@ -0,0 +1,256 @@
+
+2. Perform a **Laplacian Edge Detection kernel** on Gaussian Blurred image:
+
+::
+
+   -1 -1 -1
+   -1  8 -1
+   -1 -1 -1
+
+3. Take absolute value of Laplacian of Gaussian (LoG) result. This will quantify
+   the strength of edges with respect to the size and strength of kernel.
+4. Create a blank image, loop through each pixel and find the strongest edge in
+   the LoG(i.e. the highest value in the image stack) and take the RGB value for
+   that pixel from the corresponding image.
+
+Structural Similarity Index Measure (SSIM)
+__________________________________________
+The structural similarity index measure (SSIM) is a method for predicting the
+perceived quality of digital television and cinematic pictures, as well as other
+kinds of digital images and videos. SSIM is used for measuring the similarity
+between two images.
+
+.. image:: https://wikimedia.org/api/rest_v1/media/math/render/svg/1783c17346b8f4c822ed206798bb6769a845c417
+
+The resultant SSIM index is a decimal value between -1 and 1, and value 1 is
+only reachable in the case of two identical sets of data and therefore indicates
+perfect structural similarity. A value of 0 indicates no structural similarity.
+
+Prospective Users
+=================
+
+Macro photography or optical microscopy users who wants to take a clear image.
+They must have equipment to take multiple digital photos. This project could
+help to stack photos took with stepped focus distances.
+
+
+System Architecture
+===================
+
+**Image capture** (hardware)
+____________________________
+
+[Camera] --- PTP ---- [Control [#]_]
+
+"**Control**" Command [#]_ camera to shoot photos with stepped focus distances,
+but the hardware part is not included in this project.
+
+.. [#] Control could be PC or embedded system such as Raspberry PI.
+.. [#] This can be achived by using `Canon Digital Camera Software Development Kit (CD-SDK) <https://asia.canon/en/campaign/developerresources>`_.
+
+
+**Focus stacking** (software)
+_____________________________
+
+[**Scan for images**] ==> [**Grouping**] ==> [**Edge detect**] ==> [*Aligment* [#]_] ==> [*White Balance*] ==> [**Focus stacking**]
+
+.. [#] Aligment and white balance are not included in this project. Tested images are created and always aligned and white balanced.
+
+API Description
+===============
+
+User APIs: (Open to users)
+__________________________
+
+.. [focus_stacking] Auto scan mode: Users select specific directory that contain photos. This API
+  will scan directory and determinate the groups of photos depend on SSIM value. Then stack these
+  images together within one group.
+
+int focus_stacking(path target_path, fd* image_fd)
+  Do focus stacking in specified folder
+
+Return value:
+  0: Success
+
+  Others: Fail
+
+target_path:
+  Folder to store photos
+
+image_fd:
+  Photo file descriptors
+
+
+Private APIs (Internal used only)
+________________________________________
+
+*image gaussian_blur(image input, kernel gaussian)*
```

Comment:
You can use

```
::

  code
```

To format code snippets.

Reply:
I go through the [Quick reStructuredText](https://docutils.sourceforge.io/docs/user/rst/quickref.html#definition-lists) and using "Definition Lists"  syntax. It looks like following image.
![image](https://user-images.githubusercontent.com/9044932/95157079-a2242c80-07ca-11eb-9000-d423391f22c2.png)

Reply:
I think the signature should be changed to inline literal format, like ` ``int focus_stack()`` `.

Reply:
Thanks. I have made come changes in original branch.